src/loop/typer.ml

```diff
@@ -803,6 +803,13 @@ let non_linear_expression =
     ~hints:[text_hint]
     ~name:"Non linear expression in linear arithmetic logic" ()
 
+let bitvector_app_expected_nat =
+  Report.Error.mk ~code ~mnemonic:"bitvector-app-expected-nat"
+    ~message:(fun fmt t ->
+        Format.fprintf fmt "Expected a natural number as an argument, \
+                            instead got %a" Dolmen_std.Term.print t)
+    ~name:"Bad bitvector application argument type." ()
```

Comment:
It's not the type of the argument that's wrong.

Reply:
Fixed

Reply:
```suggestion
    ~name:"Bad bitvector application argument" ()
```